### DISCUSSION TITLE
How to optimize non-http service port requests

### DISCUSSION THREAD

**Discussion by hktalent** [projectdiscovery/nuclei#2267] [Q&A] [answered]
@ehsandeep
How to optimize non-http service port requests

If I now pass in multiple ports that are not http services, I don't want to try every http yaml in order to improve the speed of detection
http://127.0.0.1:3306
http://127.0.0.1:22

**Comment by ehsandeep** [comment, marked as answer]
@hktalent you can exclude HTTP templates with `-ept http`

**Reply by hktalent** [reply]
@ehsandeep 
But some of my goals are to use the yaml detection of the network, and adding the parameter -ept http will miss the detection of the network?

**Reply by ehsandeep** [reply]
@hktalent it will only exclude HTTP template using protocol-based filter, see the other options.

```yaml
   -pt, -type value[]                templates to run based on protocol type. Possible values: dns, file, http, headless, network, workflow, ssl, websocket, whois
   -ept, -exclude-type value[]       templates to exclude based on protocol type. Possible values: dns, file, http, headless, network, workflow, ssl, websocket, whois
```

**Reply by hktalent** [reply]
thanks
[WRN] [CVE-2018-10201] Could not execute request for http://107.182.191.202:3306: GET http://1xxx:3306/....%5C....%5C....%5C....%5C....%5C....%5C....%5C....%5C....%5Cwindows%5Cwin.ini giving up after 2 attempts: Get "http://xxxx:3306/....%5C....%5C....%5C....%5C....%5C....%5C....%5C....%5C....%5Cwindows%5Cwin.ini": net/http: HTTP/1.x transport connection broken: malformed HTTP response "J\x00\x00\x00"
......
[WRN] [lotuscms-rce] Could not execute request for http://xxxx:3306: POST http://xxx:3306/lcms/index.php giving up after 2 attempts: Post "http://xxxx:3306/lcms/index.php": net/http: HTTP/1.x transport connection broken: malformed HTTP response "J\x00\x00\x00"


I found that before all http requests, he did not confirm the correct http through the http options protocol, and began to send a large number of http requests that can be optimized and not necessary, even if the port is not http
